pkg/store/instance.go

```diff
@@ -225,3 +232,64 @@ func AddGlobalFields(inst *Instance) (FormatData, error) {
 	}
 	return data, nil
 }
+
+// PrintInstances prints instances in a requested format to a given io.Writer.
+// Supported formats are "json", "yaml", "table", or a go template
+func PrintInstances(w io.Writer, instances []*Instance, format string) error {
+	switch {
+	case format == "json":
+		b, err := json.Marshal(instances)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(w, string(b))
+	case format == "yaml":
+		b, err := yaml.Marshal(instances)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(w, string(b))
+	case format == "table":
+		w := tabwriter.NewWriter(w, 4, 8, 4, ' ', 0)
+		fmt.Fprintln(w, "NAME\tSTATUS\tSSH\tVMTYPE\tARCH\tCPUS\tMEMORY\tDISK\tDIR")
+
+		u, err := user.Current()
+		if err != nil {
+			return err
+		}
+		homeDir := u.HomeDir
+
+		for _, instance := range instances {
+			dir := instance.Dir
+			if strings.HasPrefix(dir, homeDir) {
+				dir = strings.Replace(dir, homeDir, "~", 1)
+			}
+			fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%d\t%s\t%s\t%s\n",
+				instance.Name,
+				instance.Status,
+				fmt.Sprintf("127.0.0.1:%d", instance.SSHLocalPort),
+				instance.VMType,
+				instance.Arch,
+				instance.CPUs,
+				units.BytesSize(float64(instance.Memory)),
+				units.BytesSize(float64(instance.Disk)),
+				dir,
+			)
+		}
+		return w.Flush()
+	default:
+		tmpl, err := template.New("format").Parse(format)
+		if err != nil {
+			return fmt.Errorf("invalid go template: %w", err)
+		}
```

Comment:
I still expect this to support `{{json .Foo.Bar}}`, but it can be done later

Reply:
Is that what you meant? To explicitly support it, so it can build the filtering in? OK, I get it, but, yeah, let's get this in, then do that later.